fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/DebugDrawer.kt

```diff
@@ -4,27 +4,126 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import org.mozilla.fenix.R
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
 import org.mozilla.fenix.theme.FirefoxTheme
 
 /**
  * The debug drawer UI.
+ *
+ * @param navController [NavHostController] used to perform navigation actions on the [NavHost].
+ * @param onBackButtonClick Invoked when the user taps on the back button in the app bar.
  */
 @Composable
-fun DebugDrawer() {
-    DebugDrawerHome(
-        menuItems = emptyList(),
-    )
+fun DebugDrawer(
+    navController: NavHostController,
+    onBackButtonClick: () -> Unit,
+) {
+    var backButtonVisible by remember { mutableStateOf(false) }
+    var toolbarTitle by remember { mutableStateOf("") }
+
+    // This is temporary until https://bugzilla.mozilla.org/show_bug.cgi?id=1864076
+    val homeMenuItems = List(size = 5) {
+        DebugDrawerMenuItem(
+            label = "Screen $it",
+            onClick = {
+                navController.navigate("screen_$it")
+            },
+        )
+    }
+
+    Column(modifier = Modifier.fillMaxSize()) {
+        TopAppBar(
+            title = {
+                Text(
+                    text = toolbarTitle,
+                    color = FirefoxTheme.colors.textPrimary,
+                    style = FirefoxTheme.typography.headline6,
+                )
+            },
+            navigationIcon = if (backButtonVisible) {
+                topBarBackButton(onClick = onBackButtonClick)
+            } else {
+                null
+            },
+            backgroundColor = FirefoxTheme.colors.layer1,
+            elevation = 5.dp,
+        )
+
+        NavHost(
+            navController = navController,
+            startDestination = "home",
+            modifier = Modifier.fillMaxSize(),
+        ) {
+            composable(route = "home") {
+                toolbarTitle = stringResource(id = R.string.debug_drawer_title)
+                backButtonVisible = false
+
+                DebugDrawerHome(menuItems = homeMenuItems)
+            }
+
+            homeMenuItems.forEachIndexed { index, item ->
+                composable(route = "screen_$index") {
+                    toolbarTitle = item.label
+                    backButtonVisible = true
+                    Text(
+                        text = "Screen $index",
+                        color = FirefoxTheme.colors.textPrimary,
+                        style = FirefoxTheme.typography.headline6,
+                    )
+                }
+            }
```

Comment:
This will change a little after the official logic plumb in [Bug 864076][https://bugzilla.mozilla.org/show_bug.cgi?id=1864076], but this `composable` + loop block is the magic sauce that will allow us to dynamically create the navigation graph at runtime.